impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -56,32 +59,61 @@
 
         private final List<Class<?>> interfaces;
         private final List<Class<?>> classes;
+        private final Map<String, String> classToPackageMap;
 
         private TypeInfo(Set<? extends Type> types) {
-            Comparator<Class<?>> classComparator = Comparator.comparing(Class::getName);
             List<Class<?>> foundInterfaces = new ArrayList<>();
             List<Class<?>> foundClasses = new ArrayList<>();
+            Map<String, String> classToPackage = new HashMap<>();
 
-            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses));
+            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses, classToPackage));
 
             // sort both collections and create unmodifiable lists
-            Collections.sort(foundClasses, classComparator);
-            Collections.sort(foundInterfaces, classComparator);
+            Collections.sort(foundClasses, Comparator.comparing(Class::getName));
+            Collections.sort(foundInterfaces, (o1, o2) -> {
```

Comment:
It might make sense to turn the comparator into a class and write a unit test for it ;-).

Reply:
The tests in this PR actually assert the ordering (e.g. the resulting class name).
I wanted to keep the comparator isolated in the class but I can extract it...

Reply:
It could be a nested class...

Reply:
> The tests in this PR actually assert the ordering (e.g. the resulting class name).

Yeah, I know. But a simple unit test is IMO always a good start for similar stuff. An arquillian test should always follow of course ;-).